src/main/java/org/opentripplanner/routing/api/request/request/RouteViaRequest.java

```diff
@@ -1,16 +1,162 @@
 package org.opentripplanner.routing.api.request.request;
 
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Locale;
+import java.util.function.Consumer;
+import org.opentripplanner.model.GenericLocation;
+import org.opentripplanner.routing.api.request.RouteRequest;
+import org.opentripplanner.routing.api.request.preference.RoutingPreferences;
 
-//  User request: from/to time/location; preferences slack/cost/reluctance
-
-// TODO VIA: Javadoc
+/**
+ * Trip planning request with a list of via points.
+ */
 public class RouteViaRequest {
 
-  List<ViaLocation> viaPoints;
+  private final List<ViaLeg> viaLegs = new ArrayList<>();
+
+  private GenericLocation from;
+
+  private GenericLocation to;
+
+  private Instant dateTime = Instant.now();
+
+  private Duration searchWindow;
+
+  private boolean timetableView = true;
+
+  private boolean arriveBy = false;
+
+  private Locale locale = new Locale("en", "US");
+
+  private JourneyRequest journey = new JourneyRequest();
+
+  private boolean wheelchair = false;
+
+  private Consumer<RoutingPreferences.Builder> preferences;
+
+  public RouteViaRequest(List<ViaLocation> viaLocations, List<JourneyRequest> viaJourneys) {
+    if (viaLocations == null || viaLocations.isEmpty()) {
+      throw new IllegalArgumentException("viaLocations must not be empty");
+    }
+
+    if (
+      viaJourneys == null || viaJourneys.isEmpty() || viaJourneys.size() != viaLocations.size() + 1
+    ) {
+      throw new IllegalArgumentException("There must be one more JourneyRequest than ViaLocation");
+    }
+
+    // Last ViaLeg has no ViaLocation
+    for (int i = 0; i < viaJourneys.size(); i++) {
+      var viaLocation = i < viaJourneys.size() - 1 ? viaLocations.get(i) : null;
+      viaLegs.add(new ViaLeg(viaJourneys.get(i), viaLocation));
+    }
+  }
+
+  public RouteRequest routeRequest() {
+    var request = new RouteRequest();
+
+    request.setTo(to);
+    request.setFrom(from);
+    request.setSearchWindow(searchWindow);
+    request.setDateTime(dateTime);
+    request.setArriveBy(arriveBy);
+    request.setLocale(locale);
+    request.setJourney(journey);
+    request.setWheelchair(wheelchair);
+    request.setTimetableView(timetableView);
+
+    if (preferences != null) {
+      request.withPreferences(preferences);
+    }
+
+    return request;
+  }
+
+  public void withPreferences(Consumer<RoutingPreferences.Builder> preferences) {
```

Comment:
Can we replace all of the setters with a builder pattern, as we have in many other places. That way we can validate that all required fields have been set. You can see how eg all the `*Preferences`-classes work.